realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsMongoCollection.cpp

```diff
@@ -43,47 +43,48 @@ static std::function<jobject(JNIEnv*, uint64_t)> collection_mapper_count = [](JN
 
 // This mapper works for both findOne and findOneAndUpdate/Replace functions
 static std::function<jobject(JNIEnv*, util::Optional<bson::BsonDocument>)> collection_mapper_find_one = [](JNIEnv* env, util::Optional<bson::BsonDocument> document) {
-    return document ? JniBsonProtocol::bson_to_jstring(env, *document) : NULL;
+    return document ? JniBsonProtocol::bson_to_jstring(env, *document) : nullptr;
 };
 
-static std::function<jobject(JNIEnv*, util::Optional<ObjectId>)> collection_mapper_insert_one = [](JNIEnv* env, util::Optional<ObjectId> object_id) {
-    if (object_id) {
-        return JavaClassGlobalDef::new_object_id(env, object_id.value());
+static std::function<jobject(JNIEnv*, util::Optional<Bson>)> collection_mapper_insert_one = [](JNIEnv* env, util::Optional<Bson> bson_object_id) {
+    if (bson_object_id) {
+        auto object_id = static_cast<ObjectId>(bson_object_id.value());
```

Comment:
I thought the change in ObjectStore was that this returns the primary key, which might not be an ObjectId?